AK/IpAddressCidr.h

```diff
@@ -184,8 +191,101 @@ struct Formatter<IPv4AddressCidr> : Formatter<StringView> {
 };
 #endif
 
+class IPv6AddressCidr : public Details::IPAddressCidr<IPv6AddressCidr> {
+public:
+    constexpr IPv6AddressCidr(IPv6Address address, u8 length)
+        : IPAddressCidr(address, length)
+    {
+    }
+
+    constexpr IPv6Address first_address_of_subnet() const
+    {
+        u8 address[16];
+        u8 free_bits = MAX_LENGTH - length();
+
+        if (free_bits != 128) {
+            u128 mask = NumericLimits<u128>::max() >> free_bits;
+            u8 address_mask[16];
+
+            memcpy(address_mask, &mask, sizeof(address_mask));
+
+            auto const* original_address = ip_address().to_in6_addr_t();
+            for (int i = 0; i < 16; i++) {
+                address[i] = original_address[i] & address_mask[i];
+            }
+        }
+
+        return address;
+    }
+
+    constexpr IPv6Address last_address_of_subnet() const
+    {
+        u8 inverse_address_mask[16];
+        u8 address[16] = { 0 };
+        u8 free_bits = MAX_LENGTH - length();
+        u128 inverse_mask = NumericLimits<u128>::max();
+
+        if (free_bits != 128) {
+            u8 address_mask[16];
+            u128 mask = NumericLimits<u128>::max() >> free_bits;
+            inverse_mask = inverse_mask << (128 - free_bits);
+
+            memcpy(address_mask, &mask, sizeof(address_mask));
+
+            auto const* original_address = ip_address().to_in6_addr_t();
+            for (int i = 0; i < 16; i++) {
+                address[i] = original_address[i] & address_mask[i];
+            }
+        }
```

Comment:
Same thing here, I think the other algorithm can be adopted to fill one bits instead of zero bits.